### PR TITLE
fix(ark-broker): terminal query phase lost when a reordered non-terminal event raises the sequence watermark

### LINKED ISSUE #3205: fix(ark-broker): terminal query phase permanently lost when a reordered non-terminal event raises the sequence watermark (postgres sessions)
## Problem

In the Postgres sessions backend, a query's terminal `done` phase can be permanently dropped when a non-terminal (`running`) event for the same query is applied with a higher event-stream sequence than the query's `QueryExecutionComplete` event. The `session_queries` row is then stuck at `phase='running'` forever, even though the Query CR reached `done` and the completion event was emitted. This surfaces as the intermittent `sessions-broker-postgres` e2e failure `expected 2 done session_queries rows, got '1'`.

## Context

- `applyEvent` (`services/ark-broker/ark-broker/src/brokers/sessions/postgres-sessions-storage.ts:304-325`) marks an event `stale` when its global `sequence <= session_queries.last_applied_event_sequence`, and for a stale event it skips phase application (`if (!stale) applyQueryPhase(...)`, `:325`) while still merging first-write-wins metadata. The watermark only ever climbs (`GREATEST`, `:354-357`).
- Global `sequence` is assigned at broker arrival (`http/routes/events/index.ts:88-101`, `persisted.sequenceNumber`), not per query — so arrival order need not match a single query's phase order.
- A query's `done` is carried by exactly one event (`QueryExecutionComplete` → `Done`, `session-aggregate.ts:19-20`); every other event resolves to `Running` (`:28`). `applyQueryPhase` (`session-aggregate.ts:122-148`) makes terminal phases sticky and never lets `running` overwrite an existing phase — but that precedence logic only runs when the event is *not* stale.
- The staleness guard correctly protects the terminal-vs-terminal case (a late `done` must not erase a real `error`, since `done`-clears-`error` is non-monotonic — see existing tests `postgres-sessions-storage.test.ts:771`, `:810`). The uncovered case is a **non-terminal** event raising the watermark above a not-yet-applied `done`: there is no terminal state to protect, yet `done` is discarded.
- The drop path emits no log (the only `warn` is for missing `sessionId`/`queryName`, `:264`), so it is invisible in broker logs and in `cluster-info` artifacts.
- Introduced with the feature in #3004 (2026-08-05); opt-in Postgres sessions backend; currently only on `v0.1.68-rc`. Not present on the default etcd/in-memory sessions path.

## Impact

Any query whose events reach the broker such that a non-terminal event outruns its own `QueryExecutionComplete` (reordering / concurrent OTEL export / concurrent `applyEvent` handlers under load) can be stuck non-`done` in `session_queries`. Consequences: `GET /sessions/:id` and session views under-report completed queries and over-report `running`; session status/`errorCount` aggregates skew; the e2e test flakes. Data-correctness issue for the Postgres sessions materialized view, not merely a test-timing artifact.

## Related Issues

- #3177 — same `sessions-broker-postgres` failure, filed as a ">40s propagation lag under contention" investigation. This issue identifies a deterministic root cause for that observation: the second row does not arrive late, it never arrives (permanent stale-drop). Genuine contention lag may still exist separately, but the reproduced case is not lag.
- #3074 — Phase 4 sessions review follow-ups (same subsystem).
- #3172 — widened the test poll budget (test-only mitigation); does not touch this logic.

## Reproduction

Deterministic, against a real Postgres (testcontainer). Added as a failing regression test (kept in the tree for the fix): `postgres-sessions-storage.test.ts` → `REPRO: a higher-sequence non-terminal event suppresses a later-applied done`:

1. `applyEvent(AgentExecutionStart)` at sequence **2** → row `phase=running`, watermark 2
2. `applyEvent(QueryExecutionComplete)` at sequence **1** → judged stale, phase never applied

```
Expected: "done"
Received: "running"
```

## Testing Approach

- **Unit (regression guard, present):** the repro above must resolve to `phase='done'` after the fix; existing terminal-vs-terminal ordering tests (`:771`, `:810`) must stay green (a late `done` still must not erase a real `error`).
- **Edge cases:** `done` vs later `error`/`canceled` (terminal precedence preserved); metadata first-write-wins still order-guarded independent of phase; watermark still monotonic and not poisoned to NULL by undefined sequences.
- **e2e:** `sessions-broker-postgres` passes reliably on the Postgres backend across repeated runs.

### PR DESCRIPTION
**Pull request by skazinka**
TL;DR: the Postgres session backend was allowing a "running" phase that arrived after the "done" phase (despite presumably occuring before) to override the "done", and this fixes it.

## Summary
- Fixes #3205: in the Postgres sessions backend, a query's terminal `done` phase was permanently dropped when a non-terminal (`running`) event arrived with a higher event-stream sequence than its own `QueryExecutionComplete`. `last_applied_event_sequence` advances on every event, so the later-applied terminal event looked stale and its phase was never written — leaving the `session_queries` row stuck at `running`. Surfaces as the intermittent `sessions-broker-postgres` e2e failure `expected 2 done session_queries rows, got '1'`.
- Adds `last_phase_event_sequence` (migration `000005`, additive `ADD COLUMN ... DEFAULT 0`), advanced only by terminal events, and gates phase application on it. `last_applied_event_sequence` keeps its existing jobs (metadata-redelivery suppression + status re-election) unchanged.
- Adds `isTerminalPhase` and a `touchActivity` flag to `applyQueryPhase` so a reordered terminal event writes its phase without re-electing the query as the session's latest activity.
- First commit is a deterministic regression test (real Postgres via testcontainer) that reproduces the drop; the fix flips it green. Existing terminal-vs-terminal ordering guards (a late `done` must not erase a real `error`) remain intact.
- Introduced in #3004 (Phase 4 sessions on Postgres); opt-in, rc-only. Also explains the observation in #3177 (that "propagation lag" is, at least in this case, a permanent stale-drop, not lag).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: services/ark-broker/ark-broker/src/brokers/sessions/postgres-sessions-storage.ts [hunk 1]
```diff
@@ -296,15 +298,25 @@ export class PostgresSessionsStorage implements SessionsStorage {
...
         existingRow !== undefined &&
         sequence !== undefined &&
         sequence <= Number(existingRow.last_applied_event_sequence);
+      const phaseStale =
+        existingRow !== undefined &&
+        sequence !== undefined &&
+        sequence <= Number(existingRow.last_phase_event_sequence);
+      // Only terminal events decide the phase, so only they move its watermark;
+      // GREATEST keeps it from regressing and a 0 from a non-terminal event
+      // leaves it untouched.
```
Comment (anuarora1990) [inline services/ark-broker/ark-broker/src/brokers/sessions/postgres-sessions-storage.ts:317]: do we need this comments here and above too?

**Review by giorgio-acquati-mck — APPROVED** [review]
LGTM!

File: services/ark-broker/ark-broker/src/brokers/sessions/postgres-sessions-storage.ts (hunk 1 above)
Comment (skazinka) [reply services/ark-broker/ark-broker/src/brokers/sessions/postgres-sessions-storage.ts:317]: Not sure why you hate comments so much 😆
